### DISCUSSION TITLE
.ENV files are so BUGGY

### DISCUSSION THREAD

**Discussion by bwestwood11** [vercel/next.js#52293] [Help]
### Summary

Everytime I change env files it seems it never processes the env file inside of vercel. Anybody else experiencing this problem. Super annoying.

ENV variables work in localhost and not production

**Comment by elie222** [comment]
Go to Vercel and set them in the env settings?

**Comment by bwestwood11** [comment]
Did that still happening?

**Reply by elie222** [reply]
Some ideas to help you fix it:
1. On the frontend you need to use NEXT_PUBLIC_ prefix to access. eg. NEXT_PUBLIC_MY_VAR.
2. Make sure you selected the right environments in the env section (there are three: production, preview, development). If you set the env in prod and trying to access it on a preview branch it won't work for example.
3. Redeploy the app after changing the env vars.

It's unlikely there's a bug here in Vercel.

**Reply by bwestwood11** [reply]
I have searched the internet and there is a pretty good amount of people that run into these 504 errors that time out after 10 seconds randomly.

**Comment by bwestwood11** [comment]
It only works in local host but not in production, all I did was edit the
env key in vercel. I have all three environment sections checked

**Comment by bwestwood11** [comment]
also, the file name is .env.local

**Reply by masterbater** [reply]
Go to Vercel and set them in the env settings base on @elie222 said
It doesnt read your env.local in production

**Reply by bwestwood11** [reply]
I will try that but it was working prior with .env.local even in production

**Reply by masterbater** [reply]
Its not really good to ship also your env in production or even in your repository. You should gitignore it to be safe
